src/analyze.ts

```diff
@@ -876,6 +893,46 @@ export default async function analyze(
                 pjsonPath = path.resolve(pjsonPath, '../../package.json');
               if (pjsonPath !== rootPjson) assets.add(pjsonPath);
               break;
+            case MODULE_FN:
+              if (
+                node.arguments.length > 1 &&
+                // TODO: We only cater for the case where the first argument is a string
+                node.arguments[0].type === 'Literal'
+              ) {
+                const pathOrSpecifier = node.arguments[0].value;
+                // It's a relative URL
+                if (pathOrSpecifier.startsWith('.')) {
+                  // Compute the parentURL if it's statically analyzable
+                  const computedParentURL = await computePureStaticValue(
+                    node.arguments[1],
+                  );
+
+                  if (computedParentURL && 'value' in computedParentURL) {
+                    const parentURL =
+                      computedParentURL.value instanceof URL
+                        ? computedParentURL.value.href
+                        : computedParentURL.value;
+                    // Resolve the srcURL from the parentURL
+                    const srcURL = new URL(pathOrSpecifier, parentURL).href;
+
+                    // Resolve the srcPath relative to the current file
+                    const srcPath = path.relative(
+                      path.dirname(importMetaUrl),
```

Comment:
Is it safe to call `dirname()` on a url. I thought this can only be used on paths?

Reply:
A file URL isn't much different to a Windows absolute path with a drive letter so I guess it just works?

Do the tests pass on Windows?

Is there a better alternative?

Reply:
I've removed this `dirname` usage!